pkg/archive/archive.go

```diff
@@ -0,0 +1,224 @@
+package archive
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/mholt/archiver/v3"
+	"github.com/sirupsen/logrus"
+)
+
+type Archiver interface {
```

Comment:
WDYT about using [counterfeiter](https://github.com/maxbrunsfeld/counterfeiter/) to test this?

Reply:
@estroz I was just checking out this tool and it looks really cool. I would like to do a follow on ticket to use this.